server.js

```diff
@@ -39,9 +35,22 @@ server.listen(port);
 // usernames which are currently connected to the chat
 var usernames = {};
 var data = {};
+var gameIDs = new Set();
 
+io.on('connection', function (socket) {
+  // create a new game from app.js
+  socket.on('newgame', function() {
+    var gameID = Math.round((Math.random()*10000));
```

Comment:
This is cool, something like https://www.npmjs.com/package/uuid will give you better odds at avoiding a collision or someone guessing, and prefixing the value with a millisecond-level timestamp is a way you can remove the repeated attempts to generate something unique.